src/core/qgsdistancearea.cpp

```diff
@@ -384,6 +408,13 @@ QgsPoint QgsDistanceArea::computeSpheroidProject(
   double a = mSemiMajor;
   double b = mSemiMinor;
   double f = 1 / mInvFlattening;
+  if ( ( a < 0 ) && ( b < 0 ) )
```

Comment:
This should not be required - I suspect you've needed it because you've cleared the ellipsoid parameters in setSourceCrs